common/src/stack/command/stack/commands/load/json/plugin_host.py

```diff
@@ -0,0 +1,352 @@
+# @copyright@
+# Copyright (c) 2006 - 2018 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+import json
+from stack.exception import CommandError
+
+class Plugin(stack.commands.Plugin, stack.commands.Command):
+	notifications = True
+
+	def provides(self):
+		return 'host'
+	def requires(self):
+		return [ 'software', 'environment', 'group', 'network', 'appliance', 'os', 'global' ]
+
+	def run(self, args):
+
+		# check if the user would like to import host data
+		# if there are no args, assume the user would like to import everthing
+		if args and 'host' not in args:
+			return
+
+		# self.owner.data contains the data from the json file defined in init
+		if 'host' in self.owner.data:
+			import_data = self.owner.data['host']
+		else:
+			self.owner.log.info('no host data in json file')
+			return
+
+		self.notify('\n\tLoading host\n')
+		# add each host then assign its various values to it
+		for host in import_data:
+			host_name = host['name']
+			try:
+				parameters = [
+					host_name,
+					f'box={host["box"]}',
+					f'longname={host["appliancelongname"]}',
+					f'rack={host["rack"]}',
+					f'rank={host["rank"]}'
+				]
+				if host['environment']:
+					parameters.append(f'environment={host["environment"]}')
+				self.owner.command('add.host', parameters)
+
+				self.owner.log.info(f'success adding host {host["name"]}')
+				self.owner.successes += 1
+
+			except CommandError as e:
+				if 'exists' in str(e):
+					self.owner.log.info(f'warning adding host {host["name"]}: {e}')
+					self.owner.warnings += 1
+				else:
+					self.owner.log.info(f'error adding host {host["name"]}: {e}')
+					self.owner.errors += 1
+
+
+			# iterate through each interface for the host and set it
+			for interface in host['interface']:
+				try:
+					self.owner.command('add.host.interface', [host_name, f'interface={interface["interface"]}' ])
+					self.owner.log.info(f'success adding interface {interface["interface"]}')
+					self.owner.successes += 1
+				except CommandError as e:
+					if 'exists' in str(e):
+						self.owner.log.info(f'warning adding interface {interface["interface"]}: {e}')
+						self.owner.warnings += 1
+					else:
+						self.owner.log.info(f'error adding interface {interface["interface"]}: {e}')
+						self.owner.errors += 1
+
+
+				try:
```

Comment:
By doing it this way, if one of the early conditionals raises a command error for a warning, you skip setting the rest of the interface information.  A fix for that would be looping over the keys in interface (less `interface` itself) and putting your try/except inside the loop.

We can discuss ideas for this -- it might take more creative effort than is reasonable for a Monday checkin, so I'd be willing to let this in if you'd open a ticket for it and bring it up during merge party.